tests/vmi_multus_test.go

```diff
@@ -885,15 +886,27 @@ var _ = Describe("[Serial]SRIOV", func() {
 })
 
 var _ = Describe("[Serial]Macvtap", func() {
```

Comment:
AFAIU, any test that relies on `tests.BeforeAll` cannot be parallelized, as per https://github.com/kubevirt/kubevirt/blob/d50662e2809935599300e09518a66ab6f006c479/tests/README.md#writing-e2e-tests .

Reply:
@rmohr why any tests using `tests.BeforeAll` cannot be parallelized?
In this case the `tests.BeforeAll` doesn't create any resources, it just initializes `virtClient` and calls `tests.BeforeTestCleanup()`.

Reply:
@AlonaKaplan in this case it would work. It would work in even more complex scenarios, but the issue is this: If we run the tests in parallel, let's say with 3 workers, then the tests.BeforeAll would be called three times. This is not easy to grasp and also leads to a situation where people think that they save something with a BeforeAll but it will create still things more often than they think, or they may even collide.

Reply:
@rmohr thanks for the explanation.

I understand, but if we have 9 tests in a section and 3 workers it will still save 6 calls. I agree we should be careful with using it, but not having a guideline for not using it as all.

In the network tests we introduce/change we tend to clean the resources we create in the test in the `AfterEach` of the test. Since not all the tests are using this approach we call `tests.BeforeTestCleanup()` in the beginning of the `var _` section. I think we shouldn't block our tests from being parallelized just because of this `BeforeAll`.

Reply:
@maiqueb currently the cleanup of the resources in the macvtap tests is little bit messy.
We do `tests.BeforeTestCleanup()` in the `tests.BeforeAll`.
We call `deleteAllVMIsAndBlockUntilTheyAreGone` in the `AfterEach` of some tests and delete specific vms we created in others.
Can we just call `tests.BeforeTestCleanup()` in the `BeforeEach` and remove the `tests.BeforeAll`?

Reply:
Did just that.